### PR TITLE
paginable state entity (metadata, fungibles, non fungibles, fungible …

### PR DESCRIPTION
**Pull request by PawelPawelec-RDX**
…vaults, non fungible vaults, non fungible ids) endpoints.

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.GatewayApi/Handlers/DefaultEntityHandler.cs [hunk 1]
```diff
@@ -75,84 +77,114 @@ namespace RadixDlt.NetworkGateway.GatewayApi.Handlers;
...
 
         return new GatewayModel.StateEntityDetailsResponse(ledgerState, new List<GatewayModel.StateEntityDetailsResponseItem> { item });
     }
 
-    public async Task<GatewayModel.EntityMetadataResponse?> Metadata(GatewayModel.EntityMetadataRequest request, CancellationToken token = default)
+    public async Task<GatewayModel.StateEntityMetadataPageResponse?> Metadata(GatewayModel.StateEntityMetadataPageRequest request, CancellationToken token = default)
     {
-        var ledgerState = await _ledgerStateQuerier.GetValidLedgerStateForReadRequest(request.AtLedgerState, token);
+        var ledgerState = await _ledgerStateQuerier.GetValidLedgerStateForReadRequest(
+            new GatewayModel.LedgerStateSelector { StateVersion = request.AtLedgerStateVersion }, token);
```
Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.GatewayApi/Handlers/DefaultEntityHandler.cs:105]: I'm afraid we'll have to revert this idea. :( It was a good one but the moment we decided that those pagination endpoints could be used as entry points rather continuation points only we should get this fully-fledged `at_leder_state`.

_consider this open question that does not prevent merger_

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 2]
```diff
@@ -856,31 +893,31 @@ components:
...
-        - amount
-        - last_updated_at_state_version
-      properties:
-        address:
-          $ref: "#/components/schemas/EntityAddress"
-        amount:
-          $ref: "#/components/schemas/BigDecimal"
-        last_updated_at_state_version:
-          $ref: "#/components/schemas/LastUpdatedAtStateVersion"
+#    FungibleVaultsCollection:
```
Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:896]: Why have we commented this out?

File: src/RadixDlt.NetworkGateway.GatewayApiSdk/Model/PaginableEntityCoursor.cs [hunk 3]
```diff
@@ -67,14 +67,14 @@
 namespace RadixDlt.NetworkGateway.GatewayApiSdk.Model;
 
 [DataContract]
-public sealed record EntityFungiblesCursor(int? Offset)
+public sealed record PaginableEntityCoursor(int? Offset)
```
Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.GatewayApiSdk/Model/PaginableEntityCoursor.cs:70]: I'd change the name to `OffsetCursor`. There's no guarantee that every entity-related pagination will be offset based.

**Review by KrzysztofLabus-RDX — CHANGES_REQUESTED** [review]

File: src/RadixDlt.NetworkGateway.GatewayApi/Handlers/DefaultEntityHandler.cs (hunk 1 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/Handlers/DefaultEntityHandler.cs]: fixed.

File: src/RadixDlt.NetworkGateway.GatewayApi/Handlers/DefaultEntityHandler.cs [hunk 4]
```diff
@@ -75,84 +77,114 @@ namespace RadixDlt.NetworkGateway.GatewayApi.Handlers;
...
         _entityStateQuerier = entityStateQuerier;
-    }
-
-    public async Task<GatewayModel.EntityResourcesResponse?> Resources(GatewayModel.EntityResourcesRequest request, CancellationToken token = default)
-    {
-        var ledgerState = await _ledgerStateQuerier.GetValidLedgerStateForReadRequest(request.AtLedgerState, token);
-        var aggregatePerVault = request.AggregationLevel == GatewayModel.ResourceAggregationLevel.Vault;
-
-        return await _entityStateQuerier.EntityResourcesSnapshot((GlobalAddress)request.Address, aggregatePerVault, ledgerState, token);
+        _endpointConfiguration = endpointConfiguration;
```
[the remark this answers is not in the document]
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/Handlers/DefaultEntityHandler.cs:88]: fixed.

**Review by KrzysztofLabus-RDX — APPROVED** [review]

**Pull request merged**